compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -481,6 +486,10 @@ trait ParallelTesting extends RunnerOrchestration { self =>
 
         eventualResults.foreach(_.get)
 
+        // update progress one last time
+        updateProgressMonitor()
```

Comment:
I would cancel the timer first to make sure there is no concurrent call to `updateProgressMonitor`.

Also it should only be called if `isInteractive && !suppressAllOutput` is `true`. I suggest something like:
```scala
val logProgress = isInteractive && !suppressAllOutput
if (logProgress) {
  val task = new TimerTask {
    def run() = updateProgressMonitor()
  }
  timer.schedule(task, 100, 200)
}

...

if (logProgress) {
  timer.cancel()
  // update progress one last time
  updateProgressMonitor()
}
```